tests/cupy_tests/random_tests/test_generator.py

```diff
@@ -253,6 +253,96 @@ def test_goodness_of_fit_2(self):
+    {'a': 3, 'size': 2, 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (5, 5), 'p': [0.3, 0.3, 0.4]},
+    {'a': 3, 'size': (), 'p': None},
+    {'a': [1, 2, 3], 'size': 2, 'p': [0.3, 0.3, 0.4]},
+)
+@testing.gpu
+class TestChoice(unittest.TestCase):
+
+    def setUp(self):
+        self.rs = generator.RandomState()
+
+    def test_shape(self):
+        v = self.rs.choice(a=self.a, size=self.size, p=self.p)
+        if isinstance(self.size, six.integer_types):
+            expected_shape = (self.size,)
+        else:
+            expected_shape = self.size
+        self.assertEqual(v.dtype, 'i')
+        self.assertEqual(v.shape, expected_shape)
+
+    @condition.repeat(10)
+    def test_within_choice_and_shape(self):
+        val = self.rs.choice(a=self.a, size=self.size, p=self.p).get()
+        minimum = -1 if isinstance(self.a, six.integer_types) else 0
+        maximum = 3 if isinstance(self.a, six.integer_types) else 4
+        numpy.testing.assert_array_less(
+            numpy.full(self.size, minimum, dtype=numpy.int64), val)
+        numpy.testing.assert_array_less(
+            val, numpy.full(self.size, maximum, dtype=numpy.int64))
+
+    @condition.retry(20)
+    def test_lower_bound(self):
+        val = self.rs.choice(a=self.a, size=self.size, p=self.p).get()
+        val = val.item() if self.size == () else val.item(0)
+        lower = 0 if isinstance(self.a, six.integer_types) else 1
+        self.assertEqual(lower, val)
+
+    @condition.retry(20)
+    def test_upper_bound(self):
+        val = self.rs.choice(a=self.a, size=self.size, p=self.p).get()
+        val = val.item() if self.size == () else val.item(0)
+        upper = 2 if isinstance(self.a, six.integer_types) else 3
+        self.assertEqual(upper, val)
+
+
+@testing.gpu
+class TestChoiceChi(unittest.TestCase):
+
+    def setUp(self):
+        self.rs = generator.RandomState()
+
+    def test_goodness_of_fit(self):
+        trial = 100
+        vals = [self.rs.choice(3, 1, True, [0.3, 0.3, 0.4]).get()
+                for _ in six.moves.xrange(trial)]
+        counts = numpy.histogram(vals, bins=numpy.arange(4))[0]
+        expected = numpy.array([30, 30, 40])
+        self.assertTrue(hypothesis.chi_square_test(counts, expected))
+
+    @condition.retry(5)
+    def test_goodness_of_fit_2(self):
+        vals = self.rs.choice(3, (5, 20), True, [0.3, 0.3, 0.4]).get()
+        counts = numpy.histogram(vals, bins=numpy.arange(4))[0]
+        expected = numpy.array([30, 30, 40])
+        self.assertTrue(hypothesis.chi_square_test(counts, expected))
+
+
+@testing.parameterize(
+    {'a': 3.1, 'size': 1, 'p': [0.1, 0.1, 0.8]},
+    {'a': None, 'size': 1, 'p': [0.1, 0.1, 0.8]},
+    {'a': -3, 'size': 1, 'p': [0.1, 0.1, 0.8]},
+    {'a': [[0, 1], [2]], 'size': 1, 'p': [0.1, 0.1, 0.8]},
+    {'a': 3, 'size': 1, 'p': [[0.1, 0.1], [0.8]]},
+    {'a': 2, 'size': 1, 'p': [0.1, 0.1, 0.8]},
+    {'a': 3, 'size': 1, 'p': [-0.1, 0.3, 0.8]},
+    {'a': 3, 'size': 1, 'p': [0.1, 0.1, 0.7]},
```

Comment:
Could you add a test case where `a` is an empty array for checking if [this error](https://github.com/pfnet/chainer/pull/1979/files?diff=unified#diff-bdf8bf88a45064e9d8d316ebd0d0f6fbR262) is raised appropriately.